src/apps/loyalty/containers/inquiries/index.tsx

```diff
@@ -59,42 +73,76 @@ export class Inquiries extends React.Component<RelayProps, State> {
 
     const edges = this.props.user.artwork_inquiries_connection.edges || []
     return edges.map(edge => {
-      // TODO: swap id with __id
-      const { id, artwork } = edge.node
+      const { id, artwork, impulse_conversation_id } = edge.node
       return (
-        <Col>
-          <InquiryContainer key={id}>
-            <Artwork artwork={artwork as any} />
+        <Col md={3} xs={6} key={id}>
+          <InquiryContainer>
+            <Artwork
+              artwork={artwork as any}
+              onSelect={this.onArtworkSelected.bind(this, impulse_conversation_id)}
+            />
           </InquiryContainer>
         </Col>
       )
     })
   }
 
+  onArtworkSelected(conversationId: string, selected: boolean) {
+    const selectedArtworks = this.state.selected_artworks
+    selectedArtworks[conversationId] = selected
+
+    this.setState({
+      selected_artworks: selectedArtworks,
+    })
+  }
+
   onTextboxChange(e) {
     this.setState({ self_reported_purchases: e.target.value })
   }
 
   onButtonClick(e) {
     e.preventDefault()
+    this.submitInquiriesUpdate()
+  }
 
-    const onSuccess = response => {
-      if (!response.updateCollectorProfile.loyalty_applicant_at) {
-        console.log("Loyalty Applicant Not Saved") // tslint:disable-line:no-console
-      } else {
-        console.log("Success") // tslint:disable-line:no-console
+  submitCollectorProfileUpdate() {
+    const mutation = new UpdateCollectorProfileMutation(this.state)
+
+    Relay.Store.commitUpdate(mutation, {
+      onFailure: this.onSubmitUpdatesFailed,
+      onSuccess: response => {
+        if (response.updateCollectorProfile.loyalty_applicant_at) {
+          window.location.pathname = "/loyalty/thank-you"
+        }
+      },
+    })
+  }
+
+  submitInquiriesUpdate() {
+    let ids: string[] = []
```

Comment:
Should be `const`. We’re only modifying the contents of the array, not changing the array that this variable is referring to.